['text':' Extra operator call to avoid de-duplication of the next async call','line_number':156,'multiline':False]
['text':' see https://github.com/pytorch/pytorch/pull/62710#discussion_r694680279','line_number':157,'multiline':False]
['text':' Call rpc_async from within ScriptFunction and ensure that we can attach','line_number':198,'multiline':False]
['text':' profiling callbacks. Note that handle here is a Tensor representation of','line_number':199,'multiline':False]
['text':' RecordFunction.','line_number':200,'multiline':False]
['text':' Call fork from within ScriptFunction and ensure that we can attach profiling','line_number':214,'multiline':False]
['text':' callbacks to the resulting future. Note that handle here is a Tensor','line_number':215,'multiline':False]
['text':' representation of RecordFunction.','line_number':216,'multiline':False]
['text':' create rref on current rank','line_number':256,'multiline':False]
['text':' pass rref to another user in rpc call','line_number':259,'multiline':False]
['text':' return rref in rpc call','line_number':263,'multiline':False]
['text':' pass rref to another user in remote call','line_number':267,'multiline':False]
['text':' return rref in remote call','line_number':271,'multiline':False]
['text':' pyre-ignore[7]: Pyre and torch.jit.interface don't mix well','line_number':355,'multiline':False]
['text':' Create a local RRef<MyScriptClass>.','line_number':397,'multiline':False]
['text':' Create a local RRef<MyModuleInterface>.','line_number':407,'multiline':False]
['text':' Create a local RRef<MyModuleInterface> without type hint.','line_number':412,'multiline':False]
['text':' Create a local RRef<MyScriptClass> remotely in Python.','line_number':429,'multiline':False]
['text':' Use RRef<MyScriptClass> in local Python RPC and remote Script run.','line_number':443,'multiline':False]
['text':' Use RRef<MyScriptClass> in local Script RPC and remote Script run.','line_number':447,'multiline':False]
['text':' Create a local RRef<MyModuleInterface> remotely in Python.','line_number':459,'multiline':False]
['text':' Use RRef<MyScriptClass> in local Python RPC and remote Script run.','line_number':476,'multiline':False]
['text':' Use RRef<MyScriptClass> in local Script RPC and remote Script run.','line_number':480,'multiline':False]
['text':' noqa: E999','line_number':502,'multiline':False]
['text':' Call functions remotely from Script.','line_number':540,'multiline':False]
['text':' Must annotate the value type as `Any`, because JIT type inference','line_number':603,'multiline':False]
['text':' does not support multiple types when defining a Dict.','line_number':604,'multiline':False]
['text':' The error JIT gives is,','line_number':605,'multiline':False]
['text':' "Dict values must contain only a single type, "','line_number':606,'multiline':False]
['text':' "expected: Tensor but found str instead."','line_number':607,'multiline':False]
['text':' Notice, args matching happens during scripting.','line_number':672,'multiline':False]
['text':' noqa: E999','line_number':677,'multiline':False]
['text':' Notice, args matching happens during scripting.','line_number':692,'multiline':False]
['text':' Notice, kwargs matching happens during execution.','line_number':721,'multiline':False]
['text':' noqa: E999','line_number':724,'multiline':False]
['text':' Notice, TorchScript always translates(emits) Python `raise` statement,','line_number':768,'multiline':False]
['text':' as the exception message string, "Exception",','line_number':769,'multiline':False]
['text':' no matter what exception type and exception message are in the statement,','line_number':770,'multiline':False]
['text':' create rref to itself','line_number':884,'multiline':False]
['text':' It is not thread safe to instantiate MyScriptModule in multiple threads,','line_number':905,'multiline':False]
['text':' wait for local MyScriptModule instantiation to finish,','line_number':906,'multiline':False]
['text':' otherwise it could instantiate MyScriptModule in parallel with','line_number':907,'multiline':False]
['text':' server thread in the below','line_number':908,'multiline':False]
['text':' rpc_sync still accepts script class and run it in','line_number':912,'multiline':False]
['text':' the same code path as python call.','line_number':913,'multiline':False]
['text':' rpc_sync does not accept script module method.','line_number':916,'multiline':False]
['text':' Python 3.5 and Python 3.6 throw different error message, the only','line_number':917,'multiline':False]
['text':' common word can be greped is "pickle".','line_number':918,'multiline':False]
['text':' TODO, need more investigation','line_number':926,'multiline':False]
['text':' there is rref leak when shutting down, suspect it is because','line_number':927,'multiline':False]
['text':' ref as arg is passed to pybind boundary, and the ref is not garbage','line_number':928,'multiline':False]
['text':' collected by python when calling shutdown()','line_number':929,'multiline':False]
['text':' pass rref arg to owner','line_number':942,'multiline':False]
['text':' pass rref arg to self/user','line_number':950,'multiline':False]
['text':' Construct on remote end with rpc_sync','line_number':964,'multiline':False]
['text':' Forward should output a ones tensor of self.rank.','line_number':968,'multiline':False]
['text':' Construct ScriptModule with rpc.remote.','line_number':973,'multiline':False]
['text':' Verify it is an instance of ScriptModule on remote end.','line_number':975,'multiline':False]
['text':' Run forward pass remotely.','line_number':982,'multiline':False]
['text':' Run function defined on ScriptModule remotely.','line_number':985,'multiline':False]
['text':' Ensure we can transfer ScriptModule RRef to this rank and run','line_number':988,'multiline':False]
['text':' forward pass.','line_number':989,'multiline':False]
['text':' We have no guarantee that the add_done_callback fn will execute before the test finishes.','line_number':1105,'multiline':False]
['text':' Adding a 'then' callback that runs afterwards to guarantee we wait for the first callback','line_number':1106,'multiline':False]
['text':' Ensures that we can call torch.ops.profiler._call_end_callbacks_on_jit_fut on a jit','line_number':1138,'multiline':False]
['text':' future from within a script function that calls rpc_async','line_number':1139,'multiline':False]
['text':' TODO: Can't get a reliable time for this profiling event since','line_number':1150,'multiline':False]
['text':' it's hard to estimate the execution time on the remote end for non-UDFs.','line_number':1151,'multiline':False]
['text':' This can be resolved by https://github.com/pytorch/pytorch/issues/36272.','line_number':1152,'multiline':False]
['text':' After that, this test should be modified to validate the function time.','line_number':1153,'multiline':False]
['text':' Tests that rpc_async calls made from within a TorchScript function are','line_number':1160,'multiline':False]
['text':' profiled.','line_number':1161,'multiline':False]
['text':' Ensure rpc_async call is profiled','line_number':1172,'multiline':False]
['text':' All remote events should have taken place on dst_rank','line_number':1190,'multiline':False]
['text':' script_rpc_async_call invokes add operator','line_number':1195,'multiline':False]
['text':' so we should see this as a remote event.','line_number':1196,'multiline':False]
['text':' Runs 2 rpc_async calls within JIT under record_function.','line_number':1212,'multiline':False]
['text':' Ensure record_function event is profiled.','line_number':1215,'multiline':False]
['text':' Ensure RPC future is profiled.','line_number':1222,'multiline':False]
['text':' Validate that the record_function scope time is greater than both','line_number':1233,'multiline':False]
['text':' of the individual RPC async call times. The reason it is not necessarily','line_number':1234,'multiline':False]
['text':' greater than the sum is because the two can execute in parallel.','line_number':1235,'multiline':False]
['text':' tests that torchscript functions can be profiled using with','line_number':1244,'multiline':False]
['text':' record_function(...) over RPC.','line_number':1245,'multiline':False]
['text':' Need to call below to populate CPU children.','line_number':1254,'multiline':False]
['text':' Ensures that we can call rf._call_end_callbacks_on_future on a jit','line_number':1277,'multiline':False]
['text':' future in python eager mode with torch.jit.fork','line_number':1278,'multiline':False]
['text':' Validate that callbacks were fired at the right time by checking the','line_number':1289,'multiline':False]
['text':' profiling event cpu time','line_number':1290,'multiline':False]
['text':' Ensures that we can call torch.ops.profiler._call_end_callbacks_on_jit_fut on a jit','line_number':1294,'multiline':False]
['text':' future from within a script function with torch.jit.fork','line_number':1295,'multiline':False]
['text':' @torch.jit.script complains about undefined value rpc. Error is shown','line_number':1326,'multiline':False]
['text':' below. The reason for not checking error string is to avoid making','line_number':1327,'multiline':False]
['text':' JIT error handling code depend on RPC tests, as we don't have any','line_number':1328,'multiline':False]
['text':' restrictions on the error message here.','line_number':1329,'multiline':False]
['text':'','line_number':1330,'multiline':False]
['text':' RuntimeError:','line_number':1331,'multiline':False]
['text':' undefined value rpc:','line_number':1332,'multiline':False]
['text':' def async_wrong_decorator_order(to, x, y):','line_number':1333,'multiline':False]
['text':'    # type: (str, Tensor, Tensor) -> Future[Tensor]','line_number':1334,'multiline':False]
['text':'    return rpc.rpc_async(to, script_add, (x, y))','line_number':1335,'multiline':False]
['text':'           ~~~ <--- HERE','line_number':1336,'multiline':False]
